File: lang/go/src/vac.fun/algorithms/meta/NumberTheory/NumberTheory.go
package NumberTheory

import (
    "vac.fun/algorithms/number_theory/Coprimality"
    factorial "vac.fun/algorithms/number_theory/Factorial"
    "vac.fun/algorithms/number_theory/FibonacciNumber"
    "vac.fun/algorithms/number_theory/GreatestCommonDivisor"
    "vac.fun/algorithms/number_theory/LeastCommonMultiple"
    "vac.fun/algorithms/number_theory/Parity"
    "vac.fun/algorithms/number_theory/Primality"
    "vac.fun/algorithms/number_theory/PrimeSieves"
)

func IsCoprime(m int64, n int64) bool {
    return Coprimality.ReduceToBinaryGCD(m, n)
}

func Factorial(n int64) int64 {
    return factorial.IterativeProcedure(n)
}

func Fibonacci(n int64) int64 {
    return FibonacciNumber.IterativeProcedure(n)
}

func GCD(m int64, n int64) int64 {
    return GreatestCommonDivisor.IterativeBinaryGCD(m, n)
}

func LCM(m int64, n int64) int64 {
    return LeastCommonMultiple.ReduceToBinaryGCD(m, n)
}

func IsEven(n int64) bool {
    return Parity.BitwiseIsEven(n)
}

func IsOdd(n int64) bool {
    return Parity.BitwiseIsOdd(n)
}

func IsPrime(n int64) bool {
    return Primality.IsPrime(n)
}

func IsComposite(n int64) bool {
    return Primality.IsComposite(n)
}

func SieveOfPrimes(n int) []int {
    return PrimeSieves.SieveOfEratosthenes(n)
}

File: archive/nodejs/muse/algorithms/NumberTheory.js
import * as Coprimality from './number_theory/Coprimality.js';
import * as Factorial from './number_theory/Factorial.js';
import * as FibonacciNumber from './number_theory/FibonacciNumber.js';
import * as GreatestCommonDivisor from './number_theory/GreatestCommonDivisor.js';
import * as LeastCommonMultiple from './number_theory/LeastCommonMultiple.js';
import * as Parity from './number_theory/Parity.js';
import * as Primality from './number_theory/Primality.js';
import * as PrimeSieves from './number_theory/PrimeSieves.js';

export const isCoprime = (m, n) => {
  return Coprimality.reduceToBinaryGCD(m, n);
};

export const factorial = (n) => {
  return Factorial.iterativeProcedure(n);
};

export const fibonacci = (n) => {
  return FibonacciNumber.iterativeProcedure(n);
};

export const gcd = (m, n) => {
  return GreatestCommonDivisor.iterativeBinaryGCD(m, n);
};

export const lcm = (m, n) => {
  return LeastCommonMultiple.reduceToBinaryGCD(m, n);
};

export const isEven = (n) => {
  return Parity.bitwiseIsEven(n);
};

export const isOdd = (n) => {
  return Parity.bitwiseIsOdd(n);
};

export const isPrime = (n) => {
  return Primality.isPrime(n);
};

export const isComposite = (n) => {
  return Primality.isComposite(n);
};

export const sieveOfPrimes = (n) => {
  return PrimeSieves.sieveOfEratosthenes(n);
};
